section3/reflection.md

```diff
@@ -1,13 +1,41 @@
 
 2. Think of a web or mobile application you use regularly. Where is an example of a place that the developers probably use Arrays? Explain.
+- _MyFitnessPal_ most likely uses arrays by storing related values. Each meal is an empty array. I can save what I eat for each meal and this would be added to the corresponding array. For instance, an array for breakfast would hold food like eggs, pancakes, and bacon. There would be other arrays created for lunch and dinner. If I don't log any food in, then this meal would return an empty array.
 
 3. In your own words, explain what an Array method is.
+- An **array** is a way for you to store and list multiple values inside one variable. Arrays can also store values of any data type together. There are built-in functions in Javascript that can be applied to arrays, called array methods. Array methods make developers' lives easier since we don't need to write these functions from scratch! 
 
 4. What is the purpose of a loop? When we would use a loop in conjunction with array vs. without?
+- The purpose of a **loop** is to perform repeated tasks based on a condition. A loop will keep repeating until a condition is `false`. 
+- You would use a loop **with** an array when you want to access each element individually inside the array and perform an action on that element. The condition of the loop would be based on the length of the array.
+- By itself, you can set a specific number of times you want a loop to repeat a task. You can do this **without** an array to set the condition. For example, you want to count down from 5. You set a variable for the counter and each loop will subract 1 from it:
+```
+var counter = 5;
+
+for (var i = counter; i >= 0; i--) {
+    console.log(i);
+}
+```
 
 5. If you had to teach a 5 year old about loops, how would you explain it?
 
+- A loop is like doing the same thing again and again. _Click on a dropdown for examples to give your 5 year old._
+<details>
+<summary>Example 1: Toy chest</summary>
+<br>
+You left 6 toys out of your toy chest. Let's start to clean this up. I want you to pick up one toy, walk to your empty toy chest, and put it inside. Do these steps again. You can be done when there are no toys left out. Good job!
+</details>
+<br>
+<details>
+<summary>Example 2: Block Shape Sorter</summary>
+<br>
+Let's say you have a square toy with 6 sides and empty slots. This square has heart, star, and circle slots. You need to fill the heart slot with a heart block, the start with a start block, and the circle with a circle block for this side to be complete! You move on to the next side and do the same steps. You will do this 6 times total.
+</details>
+<br>
+
 6. What questions do you still have about Arrays and loops?
+- When would you a `while` loop or `do...while` loop _instead_ of the `for` loop?
```

Comment:
Great question. I don't have a perfect answer so feel free to do your own research too!

My 2 cents: `while` and `do...while` loops are useful when you don't know the number of iterations that will be done. That said it's generally possible to do the same kind of iteration with each no matter the situation, so it often comes down to preference and readability.